miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/BoxService.java

```diff
@@ -13,14 +15,19 @@
 import uk.ac.bbsrc.tgac.miso.core.data.impl.view.BoxableView.BoxableId;
 import uk.ac.bbsrc.tgac.miso.core.util.PaginatedDataSource;
 
-public interface BoxService extends PaginatedDataSource<Box> {
+public interface BoxService extends PaginatedDataSource<Box>, BarcodableService {
+  @Override
+  public default EntityType getEntityType() {
+    return EntityType.BOX;
+  }
 
   public void deleteBox(Box box) throws IOException;
 
   public void discardAllTubes(Box box) throws IOException;
 
   public void discardSingleTube(Box box, String position) throws IOException;
 
+  @Override
   public Box get(long boxId) throws IOException;
```

Comment:
Eclipse gives the following warning in all the service classes that implement `BarcodableService`:
```
Type safety: The return type Box for get(long) from the type BoxService needs unchecked conversion to conform to T from the type BarcodableService
```

I'm not sure how to get around this, since it seems more appropriate to return a `Box` than a `<T extends Barcodable> T`.  Any suggestions are welcome.